rfc-drafts/stream.md

```diff
@@ -124,6 +124,52 @@ where
 }
 ```
 
+## Next method/struct
+
+We should also implement a next method, similar to [the implementation in the futures crate](https://docs.rs/futures-util/0.3.5/src/futures_util/stream/stream/next.rs.html#10-12).
+
+```rust
+/// Future for the [`next`](super::StreamExt::next) method.
+#[derive(Debug)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub struct Next<'a, St: ?Sized> {
+    stream: &'a mut St,
+}
```

Comment:
Because this is the first-ever proposal for inclusion of a Stream adapter in the stdlib I think it may be useful to briefly discuss the name of the trait bounds. In the stdlib [generic `Iterator` bounds are referred to by `I`](https://doc.rust-lang.org/std/iter/struct.StepBy.html). Most bounds in the stdlib are single-character, so for consistency it would be better to name the bound `S` instead of `St`.

This is what we've done in `async-std` as well, and we haven't run into any conflicts with other bounds.
```suggestion
pub struct Next<'a, S: ?Sized> {
    stream: &'a mut S,
}
```

Reply:
By the way, `futures-util` uses `St` for `Stream` because `S` may conflict with `Sink`. We use `S` in crate where `Sink` is not related, e.g., `futures-core`. It seems reasonable to use `S` here as `Sink` is not planned to be added to the standard library.

`futures-util` also use [`Fut` for `Future` because `F` may conflict with closures](https://github.com/rust-lang/futures-rs/blob/0.3.5/futures-util/src/future/future/map.rs#L13). `std` and other `futures-*` crates uses only `F`(this is related to that `std` has no utility like using both closures and futures). `async-std` uses both `Fut` and `F`, and it seems to be inconsistent overall (perhaps code directly copy-pasted from `futures-util` uses `Fut`?).